processor/transformprocessor/README.md

```diff
@@ -0,0 +1,57 @@
+# Transform Processor
+
+Supported pipeline types: traces
+
+The transform processor modifies telemetry based on configuration using the Telemetry Query Language.
+It takes a list of queries which are performed in the order specified in the config.
+
+Queries are composed of the following parts
+- Path expressions: Fields within the incoming data can be referenced using expressions composed of the names as defined
+in the OTLP protobuf definition. e.g., `status.code`, `attributes["http.method"]`. If the path expression begins with
+`resource.` or `instrumentation_library.`, it will reference those values.
+  - The name `instrumentation_library` within OpenTelemetry is currently under discussion and may be changed in the future.
+- Literals: Strings, ints, and floats can be referenced as literal values
+- Function invocations: Functions can be invoked with arguments matching the function's expected arguments
+- Where clause: Telemetry to modify can be filtered by appending `where a <op> b`, with `a` and `b` being any of the above.
+
+Supported functions:
+- `set(target, value)` - `target` is a path expression to a telemetry field to set `value` into. `value` is any value type.
+e.g., `set(attributes["http.path"], "/foo")`, `set(name, attributes["http.route"])`. If `value` resolves to `nil`, e.g.
+it references an unset map value, there will be no action.
+
+- `keep(target, string...)` - `target` is a path expression to a map type field. The map will be mutated to only contain
```

Comment:
Continuing discussion from

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6985#discussion_r777650367

```
Yes, so for example:

keep(span) where span.attribute["foo"] == "bar" -> equivalent to drop(span) where span.attribute["foo"] != "bar"

So do we decide on drop for spans with negative where statement, and keep with positive where statements for attributes?
```

_Originally posted by @bogdandrutu in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6985#discussion_r778519701_

Reply:
Hmm - I was only thinking of using `drop` for that as I haven't heard of a use case for opt-in to telemetry like that before. Make I should rename this to `keep_keys`? I don't want to overload keeping an entire item vs keeping keys from a map.

Reply:
Answer:

I think we should fail during "validation" of the config instead of "no-op". And I think is a good long-term plan, to fail fast in cases like this, especially because when you deploy the collector you want telemetry to be processed, not like in the case of an API/SDK where the telemetry is more or less a commodity.

Reply:
Sounds good. Thanks.